src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/AcceleratedDetainedAppealValidationHandler.java

```diff
@@ -0,0 +1,65 @@
+package uk.gov.hmcts.reform.iacaseapi.domain.handlers.presubmit;
+
+import static java.util.Objects.requireNonNull;
+import static uk.gov.hmcts.reform.iacaseapi.domain.entities.AsylumCaseFieldDefinition.*;
+import static uk.gov.hmcts.reform.iacaseapi.domain.entities.AsylumCaseFieldDefinition.IS_ACCELERATED_DETAINED_APPEAL;
+import static uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.field.YesOrNo.NO;
+import static uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.field.YesOrNo.YES;
+
+import java.util.Arrays;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.AsylumCase;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.DetentionFacility;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.Event;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.Callback;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.PreSubmitCallbackStage;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.field.YesOrNo;
+import uk.gov.hmcts.reform.iacaseapi.domain.handlers.PreSubmitCallbackHandler;
+
+
+@Component
+public class AcceleratedDetainedAppealValidationHandler implements PreSubmitCallbackHandler<AsylumCase> {
+
+    public boolean canHandle(
+            PreSubmitCallbackStage callbackStage,
+            Callback<AsylumCase> callback
+    ) {
+        requireNonNull(callbackStage, "callbackStage must not be null");
+        requireNonNull(callback, "callback must not be null");
+
+        return (callbackStage == PreSubmitCallbackStage.MID_EVENT)
+                && Arrays.asList(
+                Event.START_APPEAL,
```

Comment:
Do we need pageId here? Because there are many mid events which are triggered in the start and edit appeal.

Reply:
Currently in ccd-defs, there is a midEvent for isADA screen without a pageId
![image](https://github.com/hmcts/ia-case-api/assets/118450580/edde9274-b1cc-4cbb-ad34-8980780e3c68)
This implies that there may be some handlers performing logic in the backend when you press continue after selecting an option to the isADA question.

If we put a pageId in, then all the handlers currently performing logic when isADA option is chosen and continue is clicked, will stop being executed.